### PR TITLE
Catalog promotion priority

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
| --------------- | -----
| Branch?         | 1.11 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: tests/Api/Responses/Expected/admin/catalog_promotion/put_catalog_promotion_response.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "code": "mugs_discount",
     "startDate": null,
     "endDate": null,
-    "priority": 1000,
+    "priority": 0,
```
Comment (Rafikooo) [inline tests/Api/Responses/Expected/admin/catalog_promotion/put_catalog_promotion_response.json:10]: Having one row in the database when updating, gedmo sets the priority to 0, even if the value given is different

File: features/promotion/managing_catalog_promotions/seeing_catalog_promotion_details.feature [hunk 2]
```diff
@@ -12,7 +12,6 @@ Feature: Seeing catalog promotion's details
         And it applies also on "T-Shirt" product
         And it reduces also price by fixed "$10.00" in the "Web-US" channel
         And the catalog promotion "Winter sale" operates between "2021-11-10" and "2022-01-08"
-        And its priority is 1200
```
Comment (Rafikooo) [inline features/promotion/managing_catalog_promotions/seeing_catalog_promotion_details.feature:15]: The default priority value when creating is 0, which is in this scenario. This line updates the database row, but gedmo cannot allow you to set different priority values when only one row existing thus I decided to remove that assertion.
Alternatively, I can set this value to `1200` during creating, and then gedmo allows it

File: features/promotion/managing_catalog_promotions/seeing_catalog_promotion_details.feature [hunk 3]
```diff
@@ -24,4 +23,3 @@ Feature: Seeing catalog promotion's details
         And it should apply on "PHP T-Shirt" variant
         And it should apply on "T-Shirt" product
         And it should start at "2021-11-10" and end at "2022-01-08"
-        And its priority should be 1200
```
Comment (Rafikooo) [inline features/promotion/managing_catalog_promotions/seeing_catalog_promotion_details.feature:27]: With current behavior, I can assert that the priority value is `0`

File: docs/cookbook/promotions/catalog-promotion-priority.rst [hunk 4]
```diff
@@ -0,0 +1,20 @@
+How to manage catalog promotion priority?
```
Comment (GSadee) [inline docs/cookbook/promotions/catalog-promotion-priority.rst:1]: I wonder if the cookbook is a proper place for this explanation 🤔 I would be for the catalog promotion section in the book: https://docs.sylius.com/en/latest/book/products/catalog_promotions.html

File: features/promotion/managing_catalog_promotions/prioritizing_catalog_promotions.feature [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        When I create a new catalog promotion with "collection_sale" code and "Collection sale" name and 50 priority
+        Then there should be 4 catalog promotions on the list
+        And the catalog promotion named "Winter sale" should have priority 20
+        And the catalog promotion named "Year-end sale" should have priority 30
+        And the catalog promotion named "Spring sale" should have priority 40
+        And the catalog promotion named "Collection sale" should have priority 50
+
+    @api @ui
+    Scenario: Adding a catalog promotion with a priority lower than all existing ones
+              increases the priority value of other catalog promotions by 1
```
Comment (AdamKasp) [inline features/promotion/managing_catalog_promotions/prioritizing_catalog_promotions.feature:25]: I am not sure if we need these details, in the scenario we are describing what we do, not an effect, IMO assertions are enough to understand priorities logic :)

**Review by AdamKasp — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Rafał! :tada:

**Pull request merged**